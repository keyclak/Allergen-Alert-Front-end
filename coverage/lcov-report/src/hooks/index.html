
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.02% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>70/212</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/18</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">15.83% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>22/139</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">46.67% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>70/150</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="api.js"><a href="api.js.html">api.js</a></td>
	<td data-value="33.33" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 33%"></div><div class="cover-empty" style="width: 67%"></div></div>
	</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="198" class="abs low">66/198</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="16.03" class="pct low">16.03%</td>
	<td data-value="131" class="abs low">21/131</td>
	<td data-value="48.53" class="pct low">48.53%</td>
	<td data-value="136" class="abs low">66/136</td>
	</tr>

<tr>
	<td class="file low" data-value="useAsync.js"><a href="useAsync.js.html">useAsync.js</a></td>
	<td data-value="28.57" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 28%"></div><div class="cover-empty" style="width: 72%"></div></div>
	</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Fri Dec 11 2020 21:53:30 GMT-0600 (Central Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    