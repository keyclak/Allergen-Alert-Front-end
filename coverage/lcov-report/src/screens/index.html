
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">44.38% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>75/169</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">23.08% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>3/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">21.15% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>22/104</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">48.08% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>75/156</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="AddOptions.js"><a href="AddOptions.js.html">AddOptions.js</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	</tr>

<tr>
	<td class="file low" data-value="GroceryList.js"><a href="GroceryList.js.html">GroceryList.js</a></td>
	<td data-value="40" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 40%"></div><div class="cover-empty" style="width: 60%"></div></div>
	</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="15" class="abs low">6/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="15.38" class="pct low">15.38%</td>
	<td data-value="13" class="abs low">2/13</td>
	<td data-value="46.15" class="pct low">46.15%</td>
	<td data-value="13" class="abs low">6/13</td>
	</tr>

<tr>
	<td class="file high" data-value="LandingPage.js"><a href="LandingPage.js.html">LandingPage.js</a></td>
	<td data-value="85.71" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 85%"></div><div class="cover-empty" style="width: 15%"></div></div>
	</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file medium" data-value="RestrictionInfo.js"><a href="RestrictionInfo.js.html">RestrictionInfo.js</a></td>
	<td data-value="58.33" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 58%"></div><div class="cover-empty" style="width: 42%"></div></div>
	</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="24" class="abs medium">14/24</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="23.08" class="pct low">23.08%</td>
	<td data-value="13" class="abs low">3/13</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="21" class="abs medium">14/21</td>
	</tr>

<tr>
	<td class="file medium" data-value="SavedFoods.js"><a href="SavedFoods.js.html">SavedFoods.js</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	</tr>

<tr>
	<td class="file medium" data-value="SelectRestriction.js"><a href="SelectRestriction.js.html">SelectRestriction.js</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="Settings.js"><a href="Settings.js.html">Settings.js</a></td>
	<td data-value="66.67" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	</tr>

<tr>
	<td class="file medium" data-value="TypeRestriction.js"><a href="TypeRestriction.js.html">TypeRestriction.js</a></td>
	<td data-value="57.14" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	</tr>

<tr>
	<td class="file low" data-value="ViewDiet.js"><a href="ViewDiet.js.html">ViewDiet.js</a></td>
	<td data-value="23.44" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 23%"></div><div class="cover-empty" style="width: 77%"></div></div>
	</td>
	<td data-value="23.44" class="pct low">23.44%</td>
	<td data-value="64" class="abs low">15/64</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="5.56" class="pct low">5.56%</td>
	<td data-value="36" class="abs low">2/36</td>
	<td data-value="24.19" class="pct low">24.19%</td>
	<td data-value="62" class="abs low">15/62</td>
	</tr>

<tr>
	<td class="file medium" data-value="ViewFlags.js"><a href="ViewFlags.js.html">ViewFlags.js</a></td>
	<td data-value="75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 75%"></div><div class="cover-empty" style="width: 25%"></div></div>
	</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="12" class="abs medium">9/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Fri Dec 11 2020 22:49:11 GMT-0600 (Central Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    