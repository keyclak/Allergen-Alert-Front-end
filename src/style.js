import React, { useEffect, useState } from 'react';
import { View, Text, Pressable, StyleSheet, ActivityIndicator, TextInput } from 'react-native';

const Colors = {
    Foreground: '#FFF',
    Accent: '#68F',
    AccentForeground: '#FFF',
    Background: '#222',
    BackgroundBlur: '#444',
    Border: '#AAA',
    Error: '#F44',
    MainText: '#666'
};
const StyleConstants = {
    Radius: 25,
    FormItemHeight: 50,
    FormWidth: '80%',
    FormItemTextSize: 16,
    AvatarRadius: 120
};

const Styles = StyleSheet.create({
    container: {
        display: 'flex',
        height: '100%',
        flexDirection: 'column',
        alignItems: 'center',
        backgroundColor: Colors.Background
    },
    textInput: {
        height: StyleConstants.FormItemHeight,
        borderRadius: StyleConstants.Radius,
        borderWidth: 2,
        borderColor: Colors.Border,
        paddingLeft: StyleConstants.Radius,
        paddingRight: StyleConstants.Radius,
        color: Colors.Foreground,
        fontSize: StyleConstants.FormItemTextSize
    },
    button: {
        borderRadius: StyleConstants.Radius,
        height: StyleConstants.FormItemHeight,
        backgroundColor: Colors.Accent,
        alignContent: 'center',
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonText: {
        color: Colors.AccentForeground,
        fontSize: StyleConstants.FormItemTextSize
    },
    errorText: {
        color: Colors.Error,
        fontSize: StyleConstants.FormItemTextSize
    },
    labelText: {
        color: Colors.Border,
        marginLeft: StyleConstants.Radius,
        fontSize: StyleConstants.FormItemTextSize
    },
    titleText: {
        color: Colors.MainText,
        fontWeight: 'bold',
        fontSize: 20,
        paddingBottom: StyleConstants.FormItemHeight
    },
    subtitleText: {
        color: Colors.MainText,
        fontWeight: 'bold',
        fontSize: 20,
    },
    avatar: {
        width: StyleConstants.AvatarRadius,
        height: StyleConstants.AvatarRadius,
        borderRadius: 63,
        borderWidth: 2,
        borderColor: "white",
        alignSelf:'center',
        marginTop: 30, 
        marginBottom: 20
    },
    alertBox: {
        alignContent: 'center',
        justifyContent: 'center',
        color: Colors.MainText,
        fontWeight: 'bold',
        fontSize: 20,
        paddingHorizontal: 45,
        paddingVertical: 25,
        borderRadius: 20,
        backgroundColor: '#00FF99'
    },
    alert: {
        alignContent: 'center',
        justifyContent: 'center',
        color: Colors.MainText,
        fontWeight: 'bold',
        fontSize: 20,
        paddingHorizontal: 45,
        paddingVertical: 25,
        borderRadius: 20,
        backgroundColor: '#FF403D'
    },
    navButton: {
        marginBottom: 20,
        paddingVertical: 10,
        paddingHorizontal: 15,
        marginTop: 20,
        backgroundColor: "#00C4FF",
        borderRadius: 20
        },
    navButtonText: {
        fontSize: 16,
        color: "black"
        },
    buttonSpace: {
        width: 20, 
        height: 20,
        },
});

export { Colors, StyleConstants, Styles };