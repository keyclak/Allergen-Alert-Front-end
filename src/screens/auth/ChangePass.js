import React, { useContext, useState } from 'react';
import {View, Text} from 'react-native';
import TextLoadingButton from '../../components/TextLoadingButton';
import { StyleConstants, Styles } from '../../style';
import { AuthContext } from '../../context';
import FormTextInput from '../../components/FormTextInput';

export default function ChangePass({navigation}) {
    const context = useContext(AuthContext);

    const [email, setPass] = useState();

    function onReset() {
        navigation.navigate('Login');
    }

    return (
        <View style={[Styles.container]}>
            <View style={{width: StyleConstants.FormWidth}}>
            <Text style={{textAlign: 'center', fontSize: 20, fontWeight: "bold"}}>Change your password</Text>
            <Text style={{textAlign: 'center', fontSize: 18, paddingTop: 20}}>Enter a new password that hasn't been used recently</Text>
            <FormTextInput placeholder="Password"/>
            <FormTextInput placeholder="Confirm password" onChangeText={setPass}/>
            </View>
            <View style={{position: 'absolute', bottom:110, width: StyleConstants.FormWidth}}>
                <TextLoadingButton style={{ marginTop: StyleConstants.FormItemTextSize }}text="Reset Password" onPress={onReset}/>
            </View>
        </View>
    );
}