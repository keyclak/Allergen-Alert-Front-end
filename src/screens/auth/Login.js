import React, { useContext, useEffect, useState } from 'react';
import {Image, ImageBackground, View, Text, Pressable, StyleSheet, ActivityIndicator, TextInput } from 'react-native';
import LoadingButton from '../../components/LoadingButton';
import TextLoadingButton from '../../components/TextLoadingButton';
import { StyleConstants, Styles, Colors } from '../../style';
import { useLogin, useDummy } from '../../hooks/api';
import { AuthContext } from '../../context';
import FormTextInput from '../../components/FormTextInput';

export default function Login({navigation}) {
    const context = useContext(AuthContext);

    const [username, setUsername] = useState();
    const [password, setPassword] = useState();

    const login = useLogin(username, password);
    const dummy = useDummy("Hello World");

    useEffect(() => console.log(dummy.response), [dummy.response]);

    function onSubmit() {
        login.execute()
            .then(r => context.logIn(r.token))
            .catch(e => {});
    }

    function onCreateAccount() {
        navigation.navigate('CreateAccount');
    }

    function onForgotPass() {
        navigation.navigate('ForgotPass');
    }

    return (
        <View style={Styles.container}>
            <View style={{width: StyleConstants.FormWidth}}>

                <Text style={Styles.logo}>Allergen Alert</Text>

                <TextInput 
                    placeholder="Username" 
                    placeholderTextColor={Colors.Gray[5]} 
                    style={Styles.formInput}/>

                <TextInput 
                    placeholder="Password" 
                    placeholderTextColor={Colors.Gray[5]} 
                    style={Styles.formInput}/>

                <TextLoadingButton text="Log In" onPress={onSubmit} isLoading={login.loading}/>

                <Pressable style={Styles.plainButton} onPress={onForgotPass}>
                    <Text style={{color: Colors.Gray[6]}}>Forgot Password?</Text>
                </Pressable>

                <Pressable style={Styles.plainButton} onPress={onCreateAccount}>
                    <Text style={{color: Colors.Gray[5]}}>Don't have an account? </Text>
                    <Text style={{color: Colors.Gray[6]}}>Create One</Text>
                </Pressable>

            </View>
        </View>
    );
}