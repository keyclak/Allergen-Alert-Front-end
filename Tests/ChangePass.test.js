import renderer from 'react-test-renderer';
import React from 'react';
import Login from '../src/screens/auth/Login';
import { AuthContext } from '../src/context';
import MockDate from 'mockdate'
import ChangePass from '../src/screens/auth/ChangePass';

const FRAME_TIME = 10

const advanceOneFrame = () => {
  const now = Date.now();
  MockDate.set(new Date(now + FRAME_TIME));
  jest.advanceTimersByTime(FRAME_TIME);
}

const timeTravel = (msToAdvance = FRAME_TIME) => {
  const numberOfFramesToRun = msToAdvance / FRAME_TIME;
  let framesElapsed = 0;

  while (framesElapsed < numberOfFramesToRun) {
    advanceOneFrame();
    framesElapsed++;
  }
}
global.requestAnimationFrame = cb => {
  setTimeout(cb, FRAME_TIME);
}

export const setupTimeTravel = () => {
    MockDate.set(0);
    jest.useFakeTimers();
}

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.mock('@expo/vector-icons', () => ({
    MaterialIcons: 'Icon'
}));

beforeEach(setupTimeTravel);

const authContext = {
    token: null,
    logIn(token) { },
    logOut() { },
}

const mockParams = {
    token: "token",
    user: "user"
}

const route = {
    params: mockParams
}

describe("ChangePassPageTests", () => {
    test("Renders correctly", async () => {
        const tree = renderer.create(
        <AuthContext.Provider value={authContext}>
            <ChangePass route={route}/>
        </AuthContext.Provider>
        ).toJSON();

        timeTravel(500);

        expect(tree).toMatchSnapshot();
    })
})