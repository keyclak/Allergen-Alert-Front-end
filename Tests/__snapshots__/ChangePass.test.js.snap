// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassPageTests Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Change your password
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      Enter a new password that hasn't been used recently
    </Text>
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      placeholderTextColor="#868e96"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#e9ecef",
            "borderColor": "#868e96",
            "borderRadius": 15,
            "borderWidth": 0,
            "color": "#495057",
            "fontSize": 16,
            "height": 50,
            "marginBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          false,
          false,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="#868e96"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#e9ecef",
            "borderColor": "#868e96",
            "borderRadius": 15,
            "borderWidth": 0,
            "color": "#495057",
            "fontSize": 16,
            "height": 50,
            "marginBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          false,
          false,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Array [
          undefined,
          Object {
            "alignSelf": "center",
            "paddingTop": 10,
          },
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#2AB7CA",
            "borderRadius": 15,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 15,
          },
          false,
          Object {
            "marginTop": 16,
          },
        ]
      }
    >
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": 200,
              },
            ],
          }
        }
      />
      <View
        style={
          Object {
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
            }
          }
        >
          Reset Password
        </Text>
      </View>
    </View>
  </View>
</View>
`;
